admin/src/react-components/import-content.js

```diff
@@ -73,129 +80,353 @@ class ImportContentComponent extends Component {
+    if (!hadUrl) {
+      this.setState({ loadFailed: true });
     }
 
-    this.setState({
-      lastImportedUrl: this.state.url,
-      lastImportedAsset: res[type][0],
-      url: ""
-    });
+    this.setState({ urls: "", isLoading: false });
+  }
 
-    const dataProvider = window.APP.dataProvider;
-    const approveNew = isScene ? sceneApproveNew : avatarApproveNew;
-    const reviewed = isScene ? sceneReviewed : avatarReviewed;
+  async onImport(e) {
+    if (e) e.preventDefault();
+    const { imports } = this.state;
 
-    // Import API returns the sid, need to lonew_listing up the id and then check for a listing.
-    const sid = res[type][0][`${columnPrefix}_id`];
+    for (let i = 0; i < imports.length; i++) {
+      const { url, type, importUrl, isEnabled, isImported, isBase, isDefault, isFeatured } = imports[i];
+      if (isImported || !isEnabled) continue;
 
-    const objectRes = await dataProvider(GET_MANY_REFERENCE, type, {
-      sort: { field: "id", order: "desc" },
-      target: `${columnPrefix}_sid`,
-      id: sid
-    });
-    const objId = objectRes.data[0].id;
-    const listingRes = await dataProvider(GET_MANY_REFERENCE, `${columnPrefix}_listings`, {
-      sort: { field: "id", order: "desc" },
-      target: `_${columnPrefix}_id`,
-      id: objId
-    });
+      this.setImportResult(url, RESULTS.importing);
+
+      const isScene = type === "scenes";
+      const isAvatar = !isScene;
+      const columnPrefix = isScene ? "scene" : "avatar";
+      let res;
+
+      try {
+        res = await fetchReticulumAuthenticated(`/api/v1/${type}`, "POST", { url: importUrl });
+      } catch (e) {
+        this.setImportResult(url, RESULTS.failed);
+        continue;
+      }
+
+      const asset = res[type][0];
+      const tags = [];
+
+      if (isAvatar && isBase) {
+        tags.push("base");
+      }
+
+      if (isDefault) {
+        tags.push("default");
+      }
+
+      if (isFeatured) {
+        tags.push("featured");
+      }
+
+      const approveNew = isScene ? sceneApproveNew : avatarApproveNew;
+      const approveExisting = isScene ? sceneApproveExisting : avatarApproveExisting;
+      const reviewed = isScene ? sceneReviewed : avatarReviewed;
+
+      const dataProvider = window.APP.dataProvider;
+
+      // Import API returns the sid, need to look up the asset by sid and then check for a listing.
+      const sid = asset[`${columnPrefix}_id`];
 
-    // If there is an existing listing, object ends up in pending, otherwise create a listing.
-    if (listingRes.data.length === 0) {
+      const objectRes = await dataProvider(GET_MANY_REFERENCE, type, {
+        sort: { field: "id", order: "desc" },
+        target: `${columnPrefix}_sid`,
+        id: sid
+      });
+      const objId = objectRes.data[0].id;
+      const listingRes = await dataProvider(GET_MANY_REFERENCE, `${columnPrefix}_listings`, {
+        sort: { field: "id", order: "desc" },
+        target: `_${columnPrefix}_id`,
```

Comment:
Is the leading underscore a typo?

Reply:
no, this is the hacky postgrest thing we do where the `_` fields are the text-ified integer columns (this code didn't change just got moved around)